src/Microsoft.Management.Configuration.Processor/Helpers/ConfigurationUnitInternal.cs

```diff
@@ -150,6 +159,52 @@ public string ToIdentifyingString()
             return null;
         }
 
+        /// <summary>
+        /// TODO: Implement for more variables.
+        /// I am so sad because rs.SessionStateProxy.InvokeCommand.ExpandString doesn't work as I wanted.
+        /// PowerShell assumes all code passed to ExpandString is trusted and we cannot assume that.
+        /// </summary>
+        /// <returns>ValueSet with settings.</returns>
+        public ValueSet GetExpandedSettings()
+        {
+            var valueSet = new ValueSet();
+            foreach (var value in this.Unit.Settings)
+            {
+                if (value.Value is string)
+                {
+                    // For now, we just expand origin.
+                    valueSet.Add(value.Key, this.ExpandOrigin(value.Value as string, value.Key));
+                }
+                else
+                {
+                    valueSet.Add(value);
+                }
+            }
+
+            return valueSet;
+        }
+
+        private string? ExpandOrigin(string? value, string settingName)
+        {
+            if (!string.IsNullOrEmpty(value))
+            {
+                // TODO: since we only support one variable, this only finds and replace
+                // $WingetConfigRoot if found in the string when the work of expanding
+                // string is done it should take into account other operators like the subexpression operator $()
+                if (value.Contains(configRoot, StringComparison.OrdinalIgnoreCase))
+                {
+                    if (string.IsNullOrEmpty(this.origin))
+                    {
+                        throw new UnitSettingConfigRootException(this.Unit.UnitName, settingName);

```

Comment:
When will this end up happening in the overall lifetime of processing the configuration?  I would prefer that it be in response to the request to process a set, but before any actual processing takes place.

Reply:
Right now, the expansion of settings are done before right before any call to Invoke-DscResource